### ISSUE TITLE
NPE on terminating SubProcess

### ISSUE THREAD

**Issue by remcowesterhoud** [camunda-cloud/zeebe#8049]
**Describe the bug**
An NPE occurred upon termination of a sub process. This occurred in a production cluster.

https://console.cloud.google.com/errors/CMmo5-OarvWl2QE?service=zeebe&time=P7D&refresh=off&project=camunda-cloud-240911

**Expected behavior**
Sub process can be terminated without NPE

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
java.lang.NullPointerException: null
	at io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnStateBehavior.lambda$getChildInstances$0(BpmnStateBehavior.java:104) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(Unknown Source) ~[?:?]
	at io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnStateBehavior.getChildInstances(BpmnStateBehavior.java:105) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnStateTransitionBehavior.terminateChildInstances(BpmnStateTransitionBehavior.java:325) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.bpmn.container.SubProcessProcessor.onTerminate(SubProcessProcessor.java:91) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.bpmn.container.SubProcessProcessor.onTerminate(SubProcessProcessor.java:23) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processEvent(BpmnStreamProcessor.java:158) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.lambda$processRecord$1(BpmnStreamProcessor.java:131) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.util.Either$Right.ifRightOrLeft(Either.java:344) ~[zeebe-util-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processRecord(BpmnStreamProcessor.java:128) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor.processRecord(TypedRecordProcessor.java:54) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.lambda$processInTransaction$2(ProcessingStateMachine.java:258) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:84) ~[zeebe-db-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processInTransaction(ProcessingStateMachine.java:248) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processEvent(ProcessingStateMachine.java:216) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.tryToReadNextEvent(ProcessingStateMachine.java:191) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.readNextEvent(ProcessingStateMachine.java:183) ~[zeebe-workflow-engine-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.util.sched.ActorJob.invoke(ActorJob.java:73) [zeebe-util-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.util.sched.ActorTask.execute(ActorTask.java:122) [zeebe-util-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:95) [zeebe-util-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.util.sched.ActorThread.doWork(ActorThread.java:78) [zeebe-util-1.2.1.jar:1.2.1]
	at io.camunda.zeebe.util.sched.ActorThread.run(ActorThread.java:192) [zeebe-util-1.2.1.jar:1.2.1]
```

</p>
</details>

**Environment:**
 - OS: Camunda Cloud
 - Zeebe Version: 1.2.1
 - Namespace: ea41386c-4386-4d27-b988-345414a3c5fa-zeebe

**Comment by pihme** [comment]
Fixed with #8062 in `develop` and #8101 in `stable/1.2`

**Comment by Zelldon** [comment]
Is it intended to not be closed? @pihme ?
